### PR TITLE
Ambient light fixes.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
Ports a fix from Lohi's repo for ambient lights, renames STARLIGHT to AMBIENT_LIGHT, and fixes `is_outside()` behaving oddly on max-z open turfs.

## Why and what will this PR improve
Fixes ambient light corruption and turbolifts showing ambient light.

## Authorship
@Lohikar and I.

## Changelog
Nothing player-facing.

### CODE REVIEW

File: code/modules/admin/view_variables/vv_set_handlers.dm [hunk 1]
```diff
@@ -155,7 +155,7 @@
 			if (istext(value) && !color_regex.Find(value))
 				to_chat(client, "<span class='alert'><pre>ambient_light</pre> must be a 6 digit (<pre>#AABBCC</pre>) hexadecimal color string.</span>")
 				return
-			T.set_ambient_light(value)
+			T.set_ambient_light(COLOR_WHITE, value)
```
Comment (Lohikar) [inline code/modules/admin/view_variables/vv_set_handlers.dm]: This VV hook is for the first param (color).

File: code/controllers/configuration.dm [hunk 2]
```diff
@@ -203,7 +203,7 @@ var/global/list/gamemode_cache = list()
 	var/dsay_allowed = 1
 	var/aooc_allowed = 1
 
-	var/starlight = 0	// Whether space turfs have ambient light or not
+	var/ambient_light = 0	// Whether space turfs have ambient light or not
```
Comment (Lohikar) [inline code/controllers/configuration.dm]: I dunno how I feel about calling this `ambient_light`, I feel like it conflates starlight and the ambient lighting system -- which is not just for starlight and the like.

Comment (MistakeNot4892) [reply code/controllers/configuration.dm]: It's not starlight though, it's just exterior turf ambient lighting. I could name it that, I suppose.

Comment (Lohikar) [reply code/controllers/configuration.dm]: Not attached to 'starlight' specifically, just feels like some differentiation would be good.

File: code/modules/admin/view_variables/vv_set_handlers.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/modules/admin/view_variables/vv_set_handlers.dm]: So it is, not sure how I misread that.

**Review by Lohikar — DISMISSED** [review]
Per comments, the VV hook needs to be fixed.

File: code/controllers/configuration.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/controllers/configuration.dm]: Changed it to `exterior_ambient_light`.

**Comment by MistakeNot4892** [comment]
VV hook fixed.

**Review by comma — APPROVED** [review]

**Pull request merged**